src/components/Planning/GroupDialog/GroupProjectSearch.tsx

```diff
@@ -0,0 +1,141 @@
+import { IOption } from '@/interfaces/common';
+import { getProjectsWithParams } from '@/services/projectServices';
+import { arrayHasValue, listItemToOption } from '@/utils/common';
+import { Tag } from 'hds-react/components/Tag';
+import { SearchInput } from 'hds-react/components/SearchInput';
+import { FC, memo, useCallback, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Control, Controller, UseFormGetValues } from 'react-hook-form';
+import { IGroupForm } from '@/interfaces/formInterfaces';
+import { ISearchRequest } from '@/interfaces/searchInterfaces';
+
+interface IProjectSearchProps {
+  getValues: UseFormGetValues<IGroupForm>;
+  control: Control<IGroupForm, any>;
+  showAdvanceFields: boolean;
+}
+
+const GroupProjectSearch: FC<IProjectSearchProps> = ({ getValues, control, showAdvanceFields }) => {
+  const scrollRef = useRef<HTMLDivElement>(null);
+
+  const buildQueryParamString = (projectName: string): ISearchRequest => {
+    const searchParams = Object.entries(getValues())
+      .filter(([_, value]) => value?.value)
+      .map(([key, value]) => `${key}=${value.value}`);
+
+    searchParams.push(`projectName=${projectName}`);
+    searchParams.push('inGroup=false');
+    searchParams.push('programmed=true');
+
+    return { limit: '30', params: searchParams.join('&'), order: 'new' };
+  };
+
+  const { t } = useTranslation();
+  const [searchWord, setSearchWord] = useState('');
+  const [searchedProjects, setSearchedProjects] = useState<Array<IOption>>([]);
+
+  const handleValueChange = useCallback((value: string) => setSearchWord(value), []);
+  useEffect(() => {
+    if (searchedProjects.length > 0 && scrollRef.current) {
+      scrollRef.current?.scrollIntoView();
+    }
+  }, [searchedProjects]);
+
+  const getSuggestions = useCallback(
+    (inputValue: string) => {
+      if (
```

Comment:
Can you refactor the if statement that way that it will return the content of else block if true. For example:

if (true) {
  return Promise.resolve([]);
}

return new Promise....

Reply:
Done